pkg/scheduler/framework/v1alpha1/types.go

```diff
@@ -490,33 +496,44 @@ func (n *NodeInfo) AddPod(pod *v1.Pod) {
 	if affinity != nil && (affinity.PodAffinity != nil || affinity.PodAntiAffinity != nil) {
 		n.PodsWithAffinity = append(n.PodsWithAffinity, podInfo)
 	}
+	if affinity != nil && affinity.PodAntiAffinity != nil &&
+		len(affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution) != 0 {
+		n.PodsWithRequiredAntiAffinity = append(n.PodsWithRequiredAntiAffinity, podInfo)
+	}
 
 	// Consume ports when pods added.
 	n.updateUsedPorts(podInfo.Pod, true)
 
 	n.Generation = nextGeneration()
 }
 
-// RemovePod subtracts pod information from this NodeInfo.
-func (n *NodeInfo) RemovePod(pod *v1.Pod) error {
-	k1, err := GetPodKey(pod)
-	if err != nil {
-		return err
-	}
-
-	for i := range n.PodsWithAffinity {
-		k2, err := GetPodKey(n.PodsWithAffinity[i].Pod)
+func removeFromSlice(s []*PodInfo, k string) []*PodInfo {
+	for i := range s {
+		k2, err := GetPodKey(s[i].Pod)
 		if err != nil {
 			klog.Errorf("Cannot get pod key, err: %v", err)
 			continue
 		}
-		if k1 == k2 {
+		if k == k2 {
 			// delete the element
-			n.PodsWithAffinity[i] = n.PodsWithAffinity[len(n.PodsWithAffinity)-1]
-			n.PodsWithAffinity = n.PodsWithAffinity[:len(n.PodsWithAffinity)-1]
+			s[i] = s[len(s)-1]
+			s = s[:len(s)-1]
 			break
 		}
 	}
+	return s
+
+}
+
+// RemovePod subtracts pod information from this NodeInfo.
+func (n *NodeInfo) RemovePod(pod *v1.Pod) error {
+	k1, err := GetPodKey(pod)
+	if err != nil {
+		return err
+	}
+	n.PodsWithAffinity = removeFromSlice(n.PodsWithAffinity, k1)
+	n.PodsWithRequiredAntiAffinity = removeFromSlice(n.PodsWithRequiredAntiAffinity, k1)
```

Comment:
could it be optimized more by checking if the `k1` pod actually has affinity/antiaffinity before calling the remove functions?

Reply:
+1

Reply:
done.